File: 330Final-master/330Final-master/highc.go
package main
import ("fmt"
        )

func main() {

    fmt.Print("Input minimum value : ");
    var min int64 = 0;
    fmt.Scanf("%d", &min);

    fmt.Print("Input maximum value : ");
    var max int64 = 0;
    fmt.Scanf("%d", &max);
    
    var MinHighestCount int64 = 0;
    var MinseqCount int64 = 0;
    for min > 1 {
        if(min % 2 == 0) {
            min = min / 2;
        } else {
            min = 3 * min + 1;
        }
        MinseqCount = MinseqCount + 1;
        if min > MinHighestCount {
            MinHighestCount = min;
        }
    }

    var MaxHighestCount int64 = 0;
    var MaxseqCount int64 = 0;
    for max > 1 {
        if(max % 2 == 0) {
            max = max / 2;
        } else {
            max = 3 * max + 1;
        }
        MaxseqCount = MaxseqCount + 1;
        if max > MaxHighestCount {
            MaxHighestCount = max;
        }
    }

    if (MinHighestCount > MaxHighestCount) {
        fmt.Println(MinHighestCount);
    } else {
        fmt.Println(MaxHighestCount);
    }
}
    

File: 330Final-master/330Final-master/highc.f90
program highc
    integer :: minimum = 10
    integer :: maximum = 11
    integer :: MinseqCount = 0
    integer :: MaxseqCount = 0
    real :: MinHighestCount = 0
    real :: MaxHighestCount = 0

    write(*,*)("Enter the minimum and maximum value (Format: min max)")
    read (*,*)minimum,maximum

    do while(minimum > 1)
        if (modulo(minimum, 2) == 0) then
            minimum = minimum / 2
        else
            minimum = 3 * minimum + 1
        end if
        MinseqCount = MinseqCount + 1
        if (minimum > MinHighestCount) then
            MinHighestCount = minimum
        end if
    end do


    do while(maximum > 1)
        if (modulo(maximum, 2) == 0) then
            maximum = maximum / 2
        else
            maximum = 3 * maximum + 1
        end if
        MaxseqCount = MaxseqCount + 1
        if (maximum > MaxHighestCount) then
            MaxHighestCount = maximum
        end if
    end do

    if (MinHighestCount > MaxHighestCount) then
        print *, MinHighestCount
    else
        print *, MaxHighestCount
    end if

end program highc

!recursive function Collatz(num) result(HighestCount)
!    real :: seqCount
!    real :: HighestCount
!    if (num > 1) then
!        if (modulo(num, 2) == 0) then
!            num = Collatz(num / 2)
!        else
!            num = Collatz((3 * num + 1))
!        end if
!        seqCount = seqCount + 1
!        print *, seqCount
!        if (HighestCount < seqCount) then
!            HighestCount = seqCount
!        end if
!    end if
!end function Collatz

